onnxruntime/core/optimizer/fast_gelu_fusion.cc

```diff
@@ -201,6 +238,30 @@ Status FastGeluFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     input_index = optimizer_utils::IndexOfNodeInput(mul5_node, *add2_node.MutableOutputDefs()[0]);
     const Node* p_mul5_input_node = graph_utils::GetInputNode(mul5_node, (input_index + 1) % 2);
     if (p_mul5_input_node == nullptr) continue;
+
+    // if this is second formula and if pow node has Cast parent, expect mul5_node has Cast parent as well
+    NodeArg* cast_input_arg = nullptr;
+    if (second_formula) {
+      const Node* p_cast1_node = graph_utils::FirstParentByType(node, "Cast");   
+      if (p_cast1_node != nullptr) {
+        // we've done the node check in second formula for pow node
+        Node& cast1_node = *graph.GetNode(p_cast1_node->Index());
+        cast_input_arg = cast1_node.MutableInputDefs()[0];
+
+        const Node* p_cast3_node = graph_utils::FirstParentByType(mul5_node, "Cast");
+        if (p_cast3_node == nullptr) continue;
+
+        Node& cast3_node = *graph.GetNode(p_cast3_node->Index());
+        if (!CheckNode(graph, cast3_node, "Cast", {9, 13}, node.GetExecutionProviderType(), true)) {
```

Comment:
output count 1 check

Reply:
output 1 check is already covered by CheckNode() with true param

---
In reply to: [578681548](https://github.com/microsoft/onnxruntime/pull/6734#discussion_r578681548) [](ancestors = 578681548)